### PR TITLE
Convert Client to Single Page application

### PR DESCRIPTION
**Pull request by guerler**
With this PR we replace our backbone legacy router and the legacy page/layout rendering modules with Vue components and a central Vue-router. These changes convert the client application into a single page Vue application. To demonstrate the benefits the workflow editor mako has been removed, and the editor can now be opened without having to refresh the entire website.

Note that not all links within Galaxy have been rerouted to directly call the Vue-router yet, some still directly change the window location url and as such refresh the screen just as before. I think these should be changed in a follow-up, which identifies all client links and ensures that `$router.push` is called instead. Additionally the admin panel and login entry points will be incorporated into the new architecture. Overall this PR re-organizes the client components under a consistent application root component with a single router. Note also that this PR disables the legacy history route.

The home button still refreshes the entire page. This will ensure safe transition.

https://user-images.githubusercontent.com/2105447/171877718-3cb0590c-9022-4282-9b40-c272b3420f76.mov
 
## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

**Comment by bgruening** [comment]
This PR works very smoothly for me. I did not expect that it makes such a huge difference for the "feeling" to use Galaxy, but it does for me. Using the back button finally is so cool. Great work @guerler!

**Review by hujambo-dunia — APPROVED** [review]
Looked over general refactoring and pulled the branch to quickly run through locally. Definitely more performant.

File: lib/galaxy/webapps/galaxy/controllers/workflow.py [hunk 1]
```diff
@@ -740,7 +741,7 @@ def editor(self, trans, id=None, workflow_id=None, version=None):
         }
 
         # parse to mako
-        return trans.fill_template("workflow/editor.mako", editor_config=editor_config)
+        return editor_config
```
Comment (mvdbeek) [inline lib/galaxy/webapps/galaxy/controllers/workflow.py:744]: Do you want to split out the mako elimination here ? None of this controller code should survive, we should already have all you need in the regular API.

Comment (guerler) [reply lib/galaxy/webapps/galaxy/controllers/workflow.py:744]: Yes, but let's go step-by-step. I will remove the controller in a follow-up, and then continue to replace the login controller.

**Comment by dannon** [comment]
As noted this has a few issues that we can continue to iron out and it doesn't 'make galaxy a SPA', but swapping from backbone to vue router is a good, useful step.

**Pull request merged**